playbooks/adhoc/upgrades/upgrade.yml

```diff
@@ -1,35 +1,66 @@
 ---
-- name: Update deployment type
-  hosts: OSEv3
+- name: Load master facts
+  hosts: masters
   roles:
   - openshift_facts
-  post_tasks: # technically tasks are run after roles, but post_tasks is a bit more explicit.
-  - openshift_facts:
-      role: common
-      local_facts:
-        deployment_type: "{{ deployment_type }}"
 
 - name: Verify upgrade can proceed
-  hosts: masters
+  hosts: masters[0]
+  vars:
+    openshift_master_ha: "{{ groups['masters'] | length > 1 }}"
+  gather_facts: no
   tasks:
-  # Checking the global deployment type rather than host facts, this is about
-  # what the user is requesting.
-    - fail: msg="Deployment type enterprise not supported for upgrade"
+    # Checking the global deployment type rather than host facts, this is about
+    # what the user is requesting.
+    - fail:
+        msg: "Deployment type enterprise not supported for upgrade"
       when: deployment_type == "enterprise"
```

Comment:
Hmm, we're going to have bigger issues if we want to support both 3.0.z upgrades as well as 3.0.z -> 3.1 upgrades with the same playbook...